tests/lgdo/test_lh5_store.py

```diff
@@ -0,0 +1,212 @@
+    lh5_obj, n_rows = store.read_object('/data/struct/scalar', lh5_file)
+    assert isinstance(lh5_obj, lgdo.Scalar)
+    assert lh5_obj.value == 10
+    assert n_rows == 1
+
+
+def test_read_array(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/array', lh5_file)
+    assert isinstance(lh5_obj, lgdo.Array)
+    assert (lh5_obj.nda == np.array([2, 3, 4])).all()
+    assert n_rows == 3
+
+
+def test_read_vov(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/vov', lh5_file)
+    assert isinstance(lh5_obj, lgdo.VectorOfVectors)
+
+    desired = [np.array([3, 4, 5, 2]),
+               np.array([]),
+               np.array([])]
+
+    for i in range(len(desired)):
+        assert (desired[i] == list(lh5_obj)[i]).all()
+
+    assert n_rows == 3
+
+
+def test_read_aoesa(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/aoesa', lh5_file)
+    assert isinstance(lh5_obj, lgdo.ArrayOfEqualSizedArrays)
+    assert (lh5_obj.nda == np.full((3, 5), fill_value=42)).all()
+
+
+def test_read_table(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/table', lh5_file)
+    assert isinstance(lh5_obj, lgdo.Table)
+    assert n_rows == 3
+
+    # TODO: test failure here?
+    # store.read_object('/data/struct/table', [lh5_file, lh5_file])
+
+
+def test_read_lgnd_array(lgnd_file):
+    store = LH5Store()
+
+    lh5_obj, n_rows = store.read_object('/geds/raw/baseline', lgnd_file)
+    assert isinstance(lh5_obj, lgdo.Array)
+    assert n_rows == 100
+    assert len(lh5_obj) == 100
+
+    lh5_obj, n_rows = store.read_object('/geds/raw/waveform/values', lgnd_file)
+    assert isinstance(lh5_obj, lgdo.ArrayOfEqualSizedArrays)
+
+
+def test_read_array_concatenation(lgnd_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/geds/raw/baseline', [lgnd_file, lgnd_file])
+    assert isinstance(lh5_obj, lgdo.Array)
+    assert n_rows == 200
+    assert len(lh5_obj) == 200
+
+
+def test_read_lgnd_waveform_table(lgnd_file):
+    store = LH5Store()
+
+    lh5_obj, n_rows = store.read_object('/geds/raw/waveform', lgnd_file)
+    assert isinstance(lh5_obj, lgdo.WaveformTable)
+
+    lh5_obj, n_rows = store.read_object('/geds/raw/waveform', lgnd_file,
+                                        start_row=10, n_rows=10,
+                                        field_mask=['t0', 'dt'])
+
+    assert isinstance(lh5_obj, lgdo.Table)
+    assert list(lh5_obj.keys()) == ['t0', 'dt']
+    assert len(lh5_obj) == 10
+
+
+def test_lh5_iterator(lgnd_file):
+    lh5_it = LH5Iterator(lgnd_file, '/geds/raw',
+                         entry_list=range(100),
+                         field_mask=['baseline'],
+                         buffer_len=5)
+
+    lh5_obj, n_rows = lh5_it.read(4)
+    assert n_rows == 5
+    assert isinstance(lh5_obj, lgdo.Table)
+    # assert list(lh5_obj.keys()) == ['baseline'] # BUG
```

Comment:
I think there is a bug in the iterator. `field_mask` seems to have no effect.

Reply:
the bug isn't obvious. I'd like to pull these changes in and then debug.